default/python/AI/AIstate.py

```diff
@@ -386,8 +389,8 @@ def update_system_status(self):
 
             if fleet.ownedBy(empire_id):
                 if not dead_fleet:
-                    my_fleets_by_system.setdefault(this_system_id, []).append(fleet_id)
-                    fleet_spot_position.setdefault(fleet.systemID, []).append(fleet_id)
+                    self.systemStatus[this_system_id]['myfleets'].append(fleet_id)
+                    self.systemStatus[fleet.systemID]['myFleetsAccessible'].append(fleet_id)
```

Comment:
This causes an exception if, for example, the fleet is mid-starlane because it tries to access info for systemID = -1, which is not within universe.systemIDs that you initialized the dictionaries from.  Also, it is possible to observe an enemy fleet mid-starlane between systems which one has no knowledge at all about (not even basic vis) and so ```this_system_id``` might be a system ID which likewise is not within universe.systemIDs (which is only for systems known to that player)